### ISSUE TITLE
Sometimes, the detach of a kendo component does not remove all the created DOM elements, leaking memory

### ISSUE THREAD

**Issue by len-ro** [aurelia-ui-toolkits/aurelia-kendoui-bridge#721]
The problem is illustrated in the following gist: https://gist.run/?id=c127bf9e858618d26de9f3d05658848d . Add composes a template which contains a date-input, del removes the template. After removal the dom still contains the date picker popup. I played a bit with the destroy function in widget-base.js with the following results:

```
WidgetBase.prototype.destroy = function destroy(widget) {
	  widget.destroy(); //works, side effects?
      if (widget && widget.element.length > 0) {
        if (widget.wrapper && widget.wrapper !== widget.element) {
          widget.element.insertBefore(widget.wrapper);
          widget.wrapper.remove();
        }

        var classList = widget.element[0].classList;
        for (var i = 0; i < classList.length; i++) {
          var item = classList.item(i);
          if (item.startsWith('k-')) {
            classList.remove(item);
          }
        }

        widget.element.show().empty();
	//kendo.destroy(widget.element); //does not work
	  
        widget = null;

        if (this.viewModel.kWidget) {
          this.viewModel.kWidget = null;
        }
      }
    };
``` 

I do not understand the difference between kendo.destroy(widget.element) and widget.destroy(), in theory it should be the same.

**Comment by pKrav75** [comment]
Hello i ve faced a somehow similar issue showing / hidding a grid which contains combobox.
As shown in https://gist.run/?id=054958c942b939325cb807a806a4e875 when using 1.2.0 bridge, some elements remains in the dom 
![image](https://cloud.githubusercontent.com/assets/17618403/24001150/a39bd658-0a5d-11e7-9382-9ce3c926679a.png)
1.0.0Beta does not have this issue.
Chasing the difference between both version, i ve seen that .show().empty() has been added. 
Removing .empty() solves locally this particular issue.

**Comment by millman82** [comment]
I'm thinking this may actually be resolved by my pull request. #722

**Comment by JeroenVinke** [comment]
1.2.2 contains changes made in PR #722, so please confirm whether or not this fixes this issue

**Comment by len-ro** [comment]
I just tested using the 1.2.2 bundle and there are still dom elements remaining
![selection_820](https://cloud.githubusercontent.com/assets/1313020/24215040/26a356d4-0f40-11e7-8253-38a7bf2eae2e.png)

**Comment by pKrav75** [comment]
I finally used 
```javascript
if(typeof widget.destroy == 'function'){
	widget.destroy();
 }

kendo.destroy(widget.element);
widget.element.show().empty();
```
Also i m aware that widget.destroy(), and kendo.destroy(widget.element) should do the same, it does not.

This solved all my (visible) issues regarding dandling dom elements.
Does it make sense to propose this code ?

**Comment by JeroenVinke** [comment]
Thanks @pKrav75 !

**Comment by JeroenVinke** [comment]
should be fixed in 1.3.0: https://gist.run/?id=7ecd0e71fd7ef27d3954c476ed1ab264

**Comment by JeroenVinke** [comment]
@len-ro It's starting to look like that it wasn't an issue in the bridge. When I take your gist and destroy all views (instead of the latest one), then nothing gets left behind: https://gist.run/?id=c127bf9e858618d26de9f3d05658848d. 

@pKrav75 your issue seems to have been solved by https://github.com/aurelia-ui-toolkits/aurelia-kendoui-bridge/commit/3d01e65, because when I update the gist to 1.2.2 it works as expected: https://gist.run/?id=1f3fe1f982d63439c8b65209710dc1d4. 1.2.2 is before we added

```
if(typeof widget.destroy == 'function'){
	widget.destroy();
 }
```

which is causing issues now.  

@len-ro @pKrav75 I'm about to release 1.5.3 where I basically undo the change we made for this issue. I have thoroughly tested to make sure that elements are cleaned up correctly using both your examples.
